evaluator/src/main/kotlin/PackageRule.kt

```diff
@@ -0,0 +1,141 @@
+
+    override val description = "Evaluating rule '$name' for package '${pkg.id.toCoordinates()}'."
+
+    override fun runInternal() {
+        licenseRules.forEach { it.run() }
+    }
+
+    /**
+     * A [RuleMatcher] that checks if the [package][pkg] has any concluded, declared, or detected license.
+     */
+    fun hasLicense() =
+        object : RuleMatcher {
+            override val description = "hasLicense()"
+
+            override fun matches() =
+                pkg.concludedLicense?.licenses()?.isNotEmpty() == true
+                        || pkg.declaredLicenses.isNotEmpty()
+                        || detectedLicenses.isNotEmpty()
+        }
+
+    /**
+     * A [RuleMatcher] that checks if the [package][pkg] is [excluded][Excludes].
+     */
+    fun isExcluded() =
+        object : RuleMatcher {
+            override val description = "isExcluded()"
+
+            override fun matches() = ruleSet.ortResult.isExcluded(pkg.id)
+        }
+
+    /**
+     * A [RuleMatcher] that checks if the [identifier][Package.id] of the [package][pkg] belongs to one of the provided
+     * [orgs][Identifier.isFromOrg].
+     */
+    fun isFromOrg(vararg names: String) =
+        object : RuleMatcher {
+            override val description = "isFromOrg(${names.joinToString()})"
+
+            override fun matches() = pkg.id.isFromOrg(*names)
+        }
+
+    /**
+     * A [RuleMatcher] that checks if the [identifier type][Identifier.type] of the [package][pkg] equals [type].
+     */
+    fun isType(type: String) =
+        object : RuleMatcher {
+            override val description = "isType($type)"
+
+            override fun matches() = pkg.id.type == type
+        }
+
+    /**
+     * DSL function to configure a [LicenseRule] and add it to this rule.
+     */
+    fun licenseRule(name: String, licenseView: LicenseView, block: LicenseRule.() -> Unit) {
+        val licenses = licenseView.licenses(pkg, detectedLicenses.map { it.license })
+
+        licenses.forEach { (license, licenseSource) ->
+            val findings = if (licenseSource == LicenseSource.DETECTED) {
+                ruleSet.ortResult.collectLicenseFindings()[pkg.id].orEmpty()
+            } else {
+                emptyMap()
+            }
+
+            val licenseRule = LicenseRule(name, license, licenseSource, findings)
+            licenseRule.block()
+
+            licenseRules += licenseRule
+        }
+    }
+
+    /**
+     * A [Rule] to check a single license of the [package][pkg].
+     */
+    inner class LicenseRule(
+        name: String,
+
+        /**
+         * The license to check.
+         */
+        val license: String,
+
+        /**
+         * The source of the license.
+         */
+        val licenseSource: LicenseSource,
+
+        /**
+         * The associated [LicenseFinding]s. Only used if [licenseSource] is [LicenseSource.DETECTED].
+         */
+        val licenseFindings: Map<LicenseFinding, List<PathExclude>> = emptyMap()
```

Comment:
Actually, this property does not seem to be used at all, also not in `licenseRule()` where `LicenseSource.DETECTED` is used.

Reply:
It is there to be used by actual rules.